ui/core/src/api/SifService/SifService.ts

```diff
@@ -37,10 +39,15 @@ export default function createSifService({
   sifAddrPrefix,
   sifApiUrl,
   sifWsUrl,
+  keplrChainConfig,
   assets,
 }: SifServiceContext): ISifService {
   const {} = sifAddrPrefix;
-
+  const keplrProvider = getKeplrProvider()
+  // createSifService would need to be invoked at actions level for this to be cleaner imo.. maybe some sort of app initialize 
+  if(!keplrProvider) {
+    notify({type: "error", message: "Keplr not found.", detail: "Check if extension enabled for this URL."})
```

Comment:
We probably don't want to do this here. Instead you can simply create an onError handler for the service and send it up to the usecase layer and that can make the connection to notifications or notify ends up connected to everything and gets hard to test and refactor.

```ts
// In an action
api.SifService.onError((error) => {
  if(error.type === 'keplr_error') {
     notify({ ...error, type: "error"})
  }
})
```